umbraco/Umbraco-CMS
# V8: Set the form as dirty when adding/removing colors to/from Umbraco.ColorPicker

### Prerequisites


### Description

When configuring Umbraco.ColorPicker, there's no "discard changes" warning if you accidentally leave the config without saving:

![approved-color-setdirty-before](https://user-images.githubusercontent.com/7405322/61851887-613b4300-aeb8-11e9-836a-eac0fa600c5b.gif)

I ran into this issue yesterday and lost my config. So here's a PR that fixes it 😄 

_Update: This also applies to sorting of colors. As it turns out there was a bit of deprecated code going on in the `update` function for color sorting, so I removed it while I was at it._

#### Testing this PR

1. Verify that Umbraco.ColorPicker prompts you to discard changes if you add or remove colors and then leave the config without saving.
2. Verify that Umbraco.ColorPicker prompts you to discard changes if you reorder the colors and then leave the config without saving.
3. Verify that Umbraco.ColorPicker saves reordered colors correctly.
